tests/integration/security/plugin_ca_key_cert/plugin_ca_key_cert_test.go

```diff
@@ -0,0 +1,78 @@
+// Copyright 2019 Istio Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package plugincakeycert
+
+import (
+	"fmt"
+	"testing"
+	"time"
+
+	"istio.io/istio/pkg/config/constants"
+	"istio.io/istio/pkg/test/framework"
+	"istio.io/istio/pkg/test/framework/components/environment/kube"
+	"istio.io/istio/pkg/test/framework/components/istio"
+	"istio.io/istio/pkg/test/framework/components/namespace"
+	"istio.io/istio/pkg/test/framework/resource/environment"
+	"istio.io/istio/pkg/test/util/retry"
+)
+
+var (
+	testNamespace namespace.Instance
+	testCtx       framework.TestContext
+	testStruct    *testing.T
+)
+
+func TestPluginCaKeyCert(t *testing.T) {
+	framework.NewTest(t).
+		RequiresEnvironment(environment.Kube).
+		Run(func(ctx framework.TestContext) {
+			istioCfg := istio.DefaultConfigOrFail(t, ctx)
+
+			namespace.ClaimOrFail(t, ctx, istioCfg.SystemNamespace)
+			testNamespace = namespace.NewOrFail(t, ctx, namespace.Config{
+				Prefix: "test-plugin-ca-key-cert",
+				Inject: true,
+			})
+			testCtx = ctx
+			testStruct = t
+
+			// Check that the Istio CA root certificate matches the plugin CA cert.
+			retry.UntilSuccessOrFail(t, checkCACert, retry.Delay(time.Second), retry.Timeout(10*time.Second))
+		})
+}
+
+func checkCACert() error {
+	configMapName := "istio-ca-root-cert"
+	kEnv := testCtx.Environment().(*kube.Environment)
+	cm, err := kEnv.KubeClusters[0].GetConfigMap(configMapName, testNamespace.Name())
```

Comment:
I'm not sure if verifying CA cert in config map is sufficient.
The user guide shows a way to verify CA cert loaded at sidecar https://istio.io/docs/tasks/security/plugin-ca-cert/#verifying-the-certificates.
We can do similar verification in test. The goal is to cover the user flow by integration test.
There is already a test util method such as Exec() in Accessor. @myidpt

Reply:
When plugin CA certificate is used by istiod, istiod will put the CA certificate to the configmap "istio-ca-root-cert", which confirms that plugin certificate is successful.
Implementing the verification in https://istio.io/docs/tasks/security/plugin-ca-cert/#verifying-the-certificates requires a lot more development for the following items:
- Start httpbin and sleep applications; none of the existing tests have implemented starting httpbin and sleep applications.
- Retrieve the output of "openssl s-client" for httpbin.
- Parse the root certificate from the output of "openssl s-client" for httpbin.

Furthermore, as we plan to add a user guide test for https://istio.io/docs/tasks/security/plugin-ca-cert in https://github.com/istio/istio.io/tree/master/tests, which will verify CA cert loaded at sidecar https://istio.io/docs/tasks/security/plugin-ca-cert/#verifying-the-certificates. If we implement the same verification in this PR, it will be duplicating the user guide test we plan to add to https://github.com/istio/istio.io/tree/master/tests.

Reply:
It is not clear to me if CA cert in istio-ca-root-cert guarantees that CA cert is loaded to proxy as ROOTCA. It is not necessary to use httpbin, there is a test app used in integration tests. We don't need the same steps in user guide, the verification need to cover all the components. This test is run in the presubmit test and could catch code changes to CA server that break this feature.
The feature was broken once in 1.5, so we need this test to protect the flow. I will let @myidpt to assess the risks.

Reply:
Which test app to use? How to obtain and parse the CA certificate of this test app?

Reply:
Take a look at pkg/test/framework/components/echo/echo.go, it has Workloads(), and workload provides Sidecar() accessor, that provides a way to get config dump, and you can verify whether the config dump has loaded the CA cert as ROOTCA in dynamic secrets. 

There is also a Exec() method in Accessor pkg/test/kube/accessor.go, you can try if that supports executing openssl command. There are examples of using Exec() to get proxy config.

Reply:
The suggestion above involves some experiments and implementations to study its feasibility, which can be a new integration test in a separate PR.

Reply:
SGTM. Thanks!